### ISSUE TITLE
1106 error

### ISSUE THREAD

**Issue by kokos1405** [tuya/tuya-homebridge#220]
I'm getting the following error:

`TuyaOpenAPI response: {"code":1106,"msg":"permission deny","success":false,"t":1644045623936} path = /v1.0/iot-01/associated-users/actions/authorized-login
TypeError: Cannot destructure property 'access_token' of 'res.result' as it is undefined.
    at TuyaSHOpenAPI._refreshAccessTokenIfNeed (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:49:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at TuyaSHOpenAPI.request (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:145:7)
    at TuyaSHOpenAPI.getDevices (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:63:15)
    at TuyaPlatform.initTuyaSDK (/homebridge/node_modules/homebridge-tuya-platform/index.js:90:19)
Attention⚠️ ⚠️ ⚠️ ! You get an error!
Failed to get device information. Please check if the config.json is correct.`

The config is correct.The logins are correct. I have checked all the 1106 errors othes had here, but nothing can solve my problem.
Homebridge version: 1.0.4
Tuya: 1.5.0

**Comment by avielc** [comment]
This is still an issue. I receive it too
`	TuyaOpenAPI response: {"code":1106,"msg":"permission deny","success":false,"t":1649271931034,"tid":"87502123b5dc11ec83339ab42efd9e19"} path = /v1.0/iot-01/associated-users/actions/authorized-login
TypeError: Cannot destructure property 'access_token' of 'res.result' as it is undefined.
    at TuyaSHOpenAPI._refreshAccessTokenIfNeed (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:49:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at TuyaSHOpenAPI.request (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:145:7)
    at TuyaSHOpenAPI.getDevices (/homebridge/node_modules/homebridge-tuya-platform/lib/tuyashopenapi.js:63:15)
    at TuyaPlatform.initTuyaSDK (/homebridge/node_modules/homebridge-tuya-platform/index.js:90:19)
Attention⚠️ ⚠️ ⚠️ ! You get an error!
Failed to get device information. Please check if the config.json is correct.`

**Comment by megabitus98** [comment]
Hi, I had the same issue. I managed to fix mine, by entering my email address of the Tuya Account instead of the nickname in the username box.

**Comment by kareemlukitomo** [comment]
I solved error code `1106` by making sure that I was using the correct `username` and `password` in `homebridge-tuya-platform`'s JSON config.

1. Intially my password during first registration was `Password123@#`
2. Then I saw in another thead "try remove any symbol from password" so I change my auth.tuya.com account's password to `Password123`
3. After noticing the `1106` errors with the `password: Password123` in the JSON config, I tried putting in my original password in step 1 `Password123@#` and somehow it works :man_facepalming:

I am guessing cached credentials somewhere in server-side. But anyway, that was solved yesterday (on the 10th May). I haven't tried this plugin again today (assuming the cache may correct itself).

Hope it helps!
